### DISCUSSION TITLE
Relicensing k-wave-python from GPL-3.0 to LGPL-3.0: Contributor Permission Request

### DISCUSSION THREAD

**Discussion by waltsims** [waltsims/k-wave-python#525] [Announcements]
# Relicensing k-wave-python from GPL-3.0 to LGPL-3.0: Contributor Permission Request

Since the topic was raised from some of our users, we're planning to change our package's license from GPL-3.0 to LGPL-3.0 to enable broader use while maintaining core freedoms and aligning with k-wave's licensing. This change requires permission from all contributors.

## How to Give Permission
If you've contributed to this project, please comment below with:
```
I, [Your Full Name], hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.
```
Since all contributions were made through GitHub, please respond using your GitHub account that made the original contributions - this will serve as verification of contributor identity.

### Contributors Pending Response

- [x] @brubbel

### Permissions Received
[Will be updated as people respond]
- [x] @waltsims (Walter Simson)
- [x] @faberno (Fabian Schneider)
- [x] @ktritz (Kevin Tritz)
- [x] @void-mckenzie  (Mukkesh Ganesh)
- [x] @djps (David Sinden)
- [x] @talg2324
- [x] @KR616 (Kai Riemer)
- [x] @sepehrnour  (Sepehr)
- [x] @faridyagubbayli (Farid Yagubbayli)
- [x] @precicely
- [x] @ameya1252 (Ameya Deshmukh)
- [x] @gordon-n-stevenson (Gordon Stevenson)
- [x] @bvale1 (Billy Vale)
- [x] @peterhollender (Peter Hollender)
- [x] @guillefix (Guillermo Valle)

## Timeline
- Response Period: December 8th, 2024 to December 22nd, 2024
- Planned License Change: December 24th, 2024

## FAQ
**Q: Why align with k-wave's license?**
A: Since k-wave-python is a derived work of k-wave, which uses LGPL-3.0, aligning our license provides consistency in the ecosystem and clearer terms for users of both packages.

**Q: What exactly changes with LGPL-3.0?**
A: Currently under GPL-3.0, any project that imports our package must also be GPL-3.0. With LGPL-3.0, others can import and use our package while keeping their code under any license they choose. Here's what this means in practice:
```python
# This becomes allowed under any license with LGPL-3.0:
from kwave import Medium, Sensor
def user_code():
    medium = Medium()  # Only our package remains copyleft
    # User's code can be any license
```

**Q: How does this affect different Python use cases?**
A: The LGPL-3.0 license allows:
- Combining with other open source projects regardless of their license
- Using the package in commercial Python applications
- Including in internal company tools

**Q: What if someone modifies the package?**
A: Any modifications to our package's source code must still be shared under LGPL-3.0. The copyleft protection remains for our code, while allowing flexible usage of the package as a dependency.

**Q: Does this affect PyPI distribution?**
A: No, the package will still be distributed the same way through pip. Only the usage terms change, becoming more permissive.

**Q: What about existing projects using the package?**
A: This change only makes the license less restrictive, so existing projects can continue using the package as before, with more flexibility.

## Next Steps
1. We'll keep this discussion pinned until December 22nd, 2024
2. We'll follow up via email with contributors who haven't responded
3. Once we receive all permissions, we'll update the license and tag a new release

Please feel free to ask questions or raise concerns in this discussion.

**Comment by waltsims** [comment]
I, Walter Simson, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by faberno** [comment]
I, Fabian Schneider, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by ktritz** [comment]
I, Kevin Tritz, hereby give permission for my contributions to
k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I
have the legal right to grant this permission for all my contributions to
the project.

**Comment by void-mckenzie** [comment]
I, Mukkesh Ganesh, hereby give permission for my contributions to
k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I
have the legal right to grant this permission for all my contributions to
the project.

**Comment by djps** [comment]
I, David Sinden, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by talg2324** [comment]
I, Tal Grutman, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by KR616** [comment]
I, Kai Riemer, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by sepehrnour** [comment]
I, Sepehr Nourbakhsh, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by faridyagubbayli** [comment]
I, Farid Yagubbayli, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by precicely** [comment]
I, Christopher Icely, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by bvale1** [comment]
I, Billy Vale, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by guillefix** [comment]
I, Guillermo Valle, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by peterhollender** [comment]
I, Peter Hollender, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by gordon** [comment]
You tagged the wrong Gordon, you probably meant @gordon-n-stevenson.

**Comment by gordon-n-stevenson** [comment]
I, Gordon Stevenson , hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by ameya1252** [comment]
I, Ameya Deshmukh, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Comment by brubbel** [comment]
I, brubbel, hereby give permission for my contributions to k-wave-python to be relicensed from GPL-3.0 to LGPL-3.0. I confirm that I have the legal right to grant this permission for all my contributions to the project.

**Reply by brubbel** [reply]
Email sent with real name.

**Comment by waltsims** [comment]
Dear contributors,

Thank you all for approving the relicensing of k-wave-python from GPL-3.0 to LGPL-3.0. We will update the license on December 24th, 2024 with release v0.4.0.

This change allows broader usage of k-wave-python and I look forward to our future progress on the project.

Thank you,
Walter
